iaso/api/entity.py

```diff
@@ -214,6 +214,11 @@ def bulk_create(self, request, *args, **kwargs):
         serializer = EntitySerializer(entities, many=True)
         return Response(serializer.data)
 
+    def retrieve(self, request, pk=None):
+        queryset = Entity.objects.filter_for_user(self.request.user).distinct()
```

Comment:
I'm just wondering if the distinct() should not move to inside filter_for_user. Is there a reason not to do so?

Reply:
Also there is a trap with `distinct()` ([quoted from Django doc](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#django.db.models.query.QuerySet.distinct)):

> For a normal `distinct()` call, the database compares each field in each row when determining which rows are distinct.
> For a `distinct()` call with specified field names, the database will only compare the specified field names.

So I would use something like `distinct("id")` to avoid a potential performance issue.

Reply:
I'm adding the `"id"` to avoid potential issues. I've tried to add it directly inside the `filter_for_user` method, but that gives me a ` NotImplementedError: annotate() + distinct(fields) is not implemented.` on the list page.

Seems like it doesn't play well with `annotate`, so I'll just leave it on the `retrieve` action.